flytekit/core/promise.py

```diff
@@ -693,15 +693,23 @@ def create_and_link_node(
         retry_strategy or _literal_models.RetryStrategy(0),
     )
 
+    # Node ID should be the entity name, but we need to add a suffix to ensure unique node IDs
+    # when we encounter more than one such entity during compilation
+    node_id = (
```

Comment:
you have a local cluster right?  could you run through an example and add a screenshot of what things look like?  I'm worried that the `entity.name`s are too long in some cases.

Also, I was kinda hoping to gate this entire thing behind a configuration option.  Would that be possible?